samples/Gaussian_initial_state/gauss_wavefcn.qs

```diff
@@ -0,0 +1,212 @@
+namespace Microsoft.Quantum.Samples.GaussianPreparation {
+    open Microsoft.Quantum.Diagnostics;
+    open Microsoft.Quantum.Intrinsic;
+    open Microsoft.Quantum.Convert;
+    open Microsoft.Quantum.Canon;
+    open Microsoft.Quantum.Math;
+    open Microsoft.Quantum.Arrays;
+    open Microsoft.Quantum.Arithmetic;
+
+    //////////////////////////////////////////////////////////////////////////
+    // Introduction //////////////////////////////////////////////////////////
+    //////////////////////////////////////////////////////////////////////////
+
+    // This program prepares a quantum state that encodes a Gaussian function using
+    // probability amplitudes, given the standard deviation, mean, and number of 
+    // qubits.
+
+    //////////////////////////////////////////////////////////////////////////
+    // Gaussian initial state ////////////////////////////////////////////////
+    //////////////////////////////////////////////////////////////////////////
+
+    /// # Summary
+    /// Computes a single term in the normalization factor.
+    /// # Input
+    /// ## sigma
+    /// Standard deviation.
+    /// ## mu
+    /// Mean.
+    /// ## N
+    /// The term in the normalization factor.
+    function NormTerm (sigma: Double, mu: Double, N: Int) : Double {
+        let n = IntAsDouble(N);
+        return ExpD(-((n - mu) ^ 2.) /sigma ^ 2.);
+    }
+
+    /// # Summary
+    /// Computes the normalization factor.
+    /// # Input
+    /// ## sigma
+    /// Standard deviation.
+    /// ## mu
+    /// Mean.
+    /// ## N
+    /// The limit of the sum in the normalization factor.
+    function Norm (sigma: Double, mu: Double, N: Int) : Double {
+        mutable sum = 0.;
+        for (n in -N..N) {
+            set sum = PlusD(sum, NormTerm(sigma, mu, n));
+        }
+        return sum;
+    }
+
+    /// # Summary
+    /// Computes the rotation angle.
+    /// # Input
+    /// ## sigma
+    /// Standard deviation.
+    /// ## mu
+    /// Mean.
+    /// ## N
+    /// The limit of the sum in the normalization factor.
+    operation Angle (sigma: Double, mu: Double, N : Int) : Double {
+        return ArcCos(Sqrt(Norm(sigma/2., mu/2., N)/Norm(sigma, mu, N)));
+    }
+
+    /// # Summary
+    /// Return a list of n-bit strings.
+    /// # Input
+    /// ## nQubits
+    /// The number of bits.
+    operation QubitStrings (nQubits: Int) : Bool[][] {
+        return MappedOverRange(0..PowI(2, n) - 1, IntAsBoolArray(_, n));
+    }
+
+    
+    /// # Summary
+    /// Given an n-bit string, return the corresponding mean for the rotation angle
+    /// at recursion level n.
+    /// # Input
+    /// ## qub
+    /// The n-bit string.
+    /// ## mu
+    /// Mean.
+    operation MeanQubitCombo (qub: Bool[], mu: Double) : Double {
+        mutable mu_out = mu;
+        for (bit in qub) {
+            set mu_out += mu_out/2. - bit ? 0.5 | -0.5;
```

Comment:
The expression before `?` should be a Boolean expression, but this is not. Does this compile?